### PR TITLE
Removed any dependency on status-go by moving its logic to RPC calls

### PR DESCRIPTION
**Pull request by adriacidre**
Disclaimer: this is a partial request against sdk branch (this work should be moved to a different repo)

Removed any dependency on `status-go` repo, persistence and any logic managing a node.

Next steps: 
- [ ] Move `./sdk` folder to its own repo
- [ ] Correclty process [status_login response](https://github.com/status-im/status-go/compare/sdk...sdk2?expand=1#diff-bc948ae3a8d0bb6653ec85f5d702c2acR43)
- [ ] Correclty process [status_signup response](https://github.com/status-im/status-go/compare/sdk...sdk2?expand=1#diff-bc948ae3a8d0bb6653ec85f5d702c2acR56)
- [ ] Correclty process [status_joinpublicchannel response](https://github.com/status-im/status-go/compare/sdk...sdk2?expand=1#diff-bc948ae3a8d0bb6653ec85f5d702c2acR89)
- [ ] Add a proper test suite and link it on travis
- [ ] Add shortcuts for different message types `["~#c1","~#c2","~#c3","~#c4","~#c5"]`

Closes #

### CODE REVIEW

File: sdk/chan.go [hunk 1]
```diff
@@ -17,16 +17,16 @@ type Channel struct {
 
 // Publish : Publishes a message with the given body on the current channel
```
Comment (themue) [inline sdk/chan.go:18]: Would you mind changing the commenting style into typical Go style? See https://golang.org/doc/effective_go.html#commentary. It should be

```
// Publish promulgates a message with the given body on the current channel.
```

File: sdk/conn.go [hunk 2]
```diff
@@ -1,61 +1,30 @@
...
-	log.Println("Starting node...")
-
-	// Setup the storage
-	c.store = &AccountStore{}
-
-	err := statusNode.Start(config.NodeConfig)
-	if err != nil {
-		log.Fatalf("Node start failed: %v", err)
-		return err
+func New(address string) *Conn {
```
Comment (themue) [inline sdk/conn.go:18]: Would name it `NewConn()`. Calling it later with `sdk.New("1.2.3.4")` makes expecting a SDK instance. Alternatively `Conn` could be renamed to `SDK`.

File: sdk/examples/pinger/main.go [hunk 3]
```diff
@@ -8,9 +8,9 @@ import (
 )
 
 func main() {
-	conn, err := sdk.Connect("supu", "password")
-	if err != nil {
-		panic("Couldn't connect to status")
+	conn := sdk.New("rpc.server.addr:12345")
```
Comment (themue) [inline sdk/examples/pinger/main.go:11]: Here you can see how `conn := sdk.New()` feels incorrect.

**Review by themue — CHANGES_REQUESTED** [review]

**Comment by adriacidre** [comment]
Cheers @themue I'm moving this development to its own repo ATM, but i'll implement your requested those changes :)

**Pull request closed without merging**